syntax = "proto3";
package Protocol;

import "Enum.proto";
import "Struct.proto";

message GamePacket
{
	PacketHeader header = 1;
	bytes body = 2;
}

message CS_LOGIN_PACKET
{
}

message CS_INPUT_PACKET
{
	Input key = 1;
	InputPayload payload = 2;
}

message SC_LOGIN_PACKET
{
}

message SC_ADD_PACKET
{
	Vec3 pos = 1;
}

message SC_MOVE_PACKET
{
	Vec3 pos = 1;
	float rot = 2;
}

message SC_REMOVE_PACKET
{
}

message SC_ATTACK_PACKET
{
	float rot = 1;
}

message SC_DODGE_PACKET
{
	Vec3 dir = 1;
}